benchmarks/_benchmark.py

```diff
@@ -58,13 +58,29 @@ class ConbenchCommunicator(conbench.runner.Conbench):
     publish() method overridden to use the new retrying client.
     """
 
-    def publish(self, benchmark: dict) -> None:
-        # Put login information into environment variables so the new client can access it
+    # Upon first initialization, cache the new retrying client here so we don't have to
+    # login over and over.
+    _conbench_client: Optional[ConbenchClient] = None
+
+    @property
+    def conbench_client(self) -> ConbenchClient:
+        """Set up the new retrying ConbenchClient. And attempt to login.
+
+        (The new client needs login information in environment variables.)
+        """
+        if self._conbench_client:
+            return self._conbench_client
+
         os.environ["CONBENCH_URL"] = self.config.login_url.split("/api/login")[0]
         os.environ["CONBENCH_EMAIL"] = self.config.credentials["email"]
         os.environ["CONBENCH_PASSWORD"] = self.config.credentials["password"]
 
-        ConbenchClient().post("/api/benchmarks", benchmark)
+        # Login happens here.
+        self._conbench_client = ConbenchClient()
+        return self._conbench_client
+
+    def publish(self, benchmark: dict) -> None:
+        self.conbench_client.post("/api/benchmark-results", benchmark)
```

Comment:
That `ConbenchClient` that we replaced here must have been a different ConbenchClient from the one we replaced before in txxxx-tools.

See pull/394/files#diff-d0b70ea9d871ec6f89c3d337dc64ab58ba6c768beb91f6a82f3eae4bbae257eeR449 where the old interface usage was 

```
conbench_client.post("/benchmarks/", json=benchmark_result_json)
```
and here it was
```
ConbenchClient().post("/api/benchmarks", benchmark)
```

Important difference w.r.t. to the url path prefix.

Fallout: https://github.com/voltrondata-labs/arrow-benchmarks-ci/issues/128